docs/Endpoints.md

```diff
@@ -117,43 +117,41 @@ provide compile-time checking of well-defined network targets. You've already
 seen how to map targets into endpoints using the `endpointClosure` parameter
 of the `MoyaProvider` initializer. That lets you create an `Endpoint` instance
 that Moya will use to reason about the network API call. At some point, that
-`Endpoint` must be resolved into an actual `NSURLRequest` to give to Alamofire.
+`Endpoint` must be resolved into an actual `URLRequest` to give to Alamofire.
 That's what the `requestClosure` parameter is for.
 
 The `requestClosure` is an optional, last-minute way to modify the request
 that hits the network. It has a default value of `MoyaProvider.DefaultRequestMapper`,
 which simply uses the `urlRequest` property of the `Endpoint` instance.
 
 This closure receives an `Endpoint` instance and is responsible for invoking a
-its argument of `RequestResultClosure` (shorthand for `Result<NSURLRequest, Moya.Error> -> Void`) with a request that represents the Endpoint.
+its argument of `RequestResultClosure` (shorthand for `Result<URLRequest, Moya.Error> -> Void`) with a request that represents the Endpoint.
 It's here that you'd do your OAuth signing or whatever. Since you may invoke the
 closure asynchronously, you can use whatever authentication library you like ([example](https://github.com/rheinfabrik/Heimdallr.swift)).
 Instead of modifying the request, you could simply log it, instead.
 
 ```swift
 let requestClosure = { (endpoint: Endpoint<GitHub>, done: MoyaProvider.RequestResultClosure) in
-    // Using the `as!` forced type cast operator is safe here,
-    // as `mutableCopy()` will always return the correct type.
-    let request = endpoint.urlRequest.mutableCopy() as! NSMutableURLRequest
+    var request = endpoint.urlRequest
```

Comment:
With the new value types, there's no need to use NSMutableRequest. You instead use a `var` 💃